### PR TITLE
Space X Defense Program Haifeng Su

### PR DESCRIPTION
**Pull request by SuHaifeng**
-Added Projectiles
-Fixed data Structures
-misc

### CODE REVIEW

File: Space-X Defense Program/Space-X Defense Program/CGPoint+VectorMath.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
+//
+//  CGPoint+VectorMath.swift
+//  Space-X Defense Program
+//
+//  Created by Su Haifeng on 3/19/18.
```
Comment (k-kruusi) [inline Space-X Defense Program/Space-X Defense Program/CGPoint+VectorMath.swift:5]: If your just going to copy my code line for line do you really need to attribute it to yourself?

File: Space-X Defense Program/Space-X Defense Program/Player.swift [hunk 2]
```diff
@@ -30,9 +30,11 @@ class Player: GameObject {
         fatalError("init(coder:) has not been implemented")
     }
     
-    /// update method to inject updates into the crazy catlady during the game loop
-    ///
-    /// - Parameter currentTime: the current time
+    func posY () -> CGFloat
```
Comment (k-kruusi) [inline Space-X Defense Program/Space-X Defense Program/Player.swift:33]: ok...

File: Space-X Defense Program/Space-X Defense Program/StructValues.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+//  Space-X Defense Program
+//
+//  Created by Su Haifeng on 3/19/18.
+//  Copyright © 2018 Su Haifeng. All rights reserved.
+//
+
+import Foundation
+import SpriteKit
+
+
```
Comment (k-kruusi) [inline Space-X Defense Program/Space-X Defense Program/StructValues.swift:12]: 👍

File: Space-X Defense Program/Space-X Defense Program/CGPoint+VectorMath.swift (hunk 1 above)
Comment (SuHaifeng) [reply Space-X Defense Program/Space-X Defense Program/CGPoint+VectorMath.swift:5]: Sorry about that~ when I created a new swift file ~ it automatically placed my name~ I will change this in the future~

Comment (kevin-kruusi) [reply Space-X Defense Program/Space-X Defense Program/CGPoint+VectorMath.swift:5]: Check this out so you better understand. https://choosealicense.com/no-permission/ Im your teacher and its ok but sometimes even if you see an example your not aloud to use it verbatim.

**Pull request merged**